src/msg/producer/writer/message.go

```diff
@@ -61,6 +58,7 @@ func (m *message) Set(meta metadata, rm *producer.RefCountedMessage, initNanos i
 
 // Close resets the states of the message.
 func (m *message) Close() {
+	m.RefCountedMessage = nil
```

Comment:
Just for my understanding--this is the memory leak you were referring to right?

Reply:
yes, albeit small, it is a leak.
unlike RefCountedMessage, there are no calls Close() to drop message out of band or anything like that.